File: 0414-third-maximum-number/0414-third-maximum-number.java
class Solution {
    public int thirdMax(int[] nums) {
        Long max1 = null;
        Long max2 = null;
        Long max3 = null;

        for (int n : nums) {
            
            if ((max1 != null && n == max1) ||
                (max2 != null && n == max2) ||
                (max3 != null && n == max3)) {
                continue;
            }

            if (max1 == null || n > max1) {
                max3 = max2;
                max2 = max1;
                max1 = (long) n;
            } 
            else if (max2 == null || n > max2) {
                max3 = max2;
                max2 = (long) n;
            } 
            else if (max3 == null || n > max3) {
                max3 = (long) n;
            }
        }

        return max3 == null ? max1.intValue() : max3.intValue();
    }
}

File: 0414-third-maximum-number/0414-third-maximum-number.py
class Solution:
    def thirdMax(self, nums: list[int]) -> int:
        n=len(nums)
        first=float('-inf')
        second=float('-inf')
        third=float('-inf')
        for i in nums:
            if i in (first,second,third):
                continue
                
            if i > first:
                third=second
                second=first
                first=i
            elif  i > second:
                third=second
                second=i
            elif i > third:
                third=i
            
        return third if third !=float('-inf') else first

        